src/modules/configuration/src/lib/Configuration.c

```diff
@@ -0,0 +1,547 @@
+    }
+
+    if (MMI_OK == status)
+    {
+        *payloadSizeBytes = strlen(buffer);
+
+        if ((g_maxPayloadSizeBytes > 0) && ((unsigned)*payloadSizeBytes > g_maxPayloadSizeBytes))
+        {
+            OsConfigLogError(ConfigurationGetLog(), "MmiGet(%s, %s) insufficient maxmimum size (%d bytes) versus data size (%d bytes), reported buffer will be truncated",
+                componentName, objectName, g_maxPayloadSizeBytes, *payloadSizeBytes);
+
+            *payloadSizeBytes = g_maxPayloadSizeBytes;
+        }
+
+        *payload = (MMI_JSON_STRING)malloc(*payloadSizeBytes);
+        if (*payload)
+        {
+            memcpy(*payload, buffer, *payloadSizeBytes);
+        }
+        else
+        {
+            OsConfigLogError(ConfigurationGetLog(), "MmiGet: failed to allocate %d bytes", *payloadSizeBytes + 1);
+            *payloadSizeBytes = 0;
+            status = ENOMEM;
+        }
+    }    
+
+    if (IsFullLoggingEnabled())
+    {
+        OsConfigLogInfo(ConfigurationGetLog(), "MmiGet(%p, %s, %s, %.*s, %d) returning %d", clientSession, componentName, objectName, *payloadSizeBytes, *payload, *payloadSizeBytes, status);
+    }
+
+    FREE_MEMORY(configuration);
+
+    return status;
+}
+
+int ConfigurationMmiSet(MMI_HANDLE clientSession, const char* componentName, const char* objectName, const MMI_JSON_STRING payload, const int payloadSizeBytes)
+{
+    int status = MMI_OK;
+
+    char* payloadString = NULL;
+
+    int modelVersion = g_modelVersion;
+    int refreshInterval = g_refreshInterval;
+    int localManagementEnabled = g_localManagementEnabled ? 1 : 0;
+    int fullLoggingEnabled = g_fullLoggingEnabled ? 1 : 0;
+    int commandLoggingEnabled = g_commandLoggingEnabled ? 1 : 0;
+    int iotHubProtocol = g_iotHubProtocol;
+
+    JSON_Value* jsonValue = NULL;
+    JSON_Object* jsonObject = NULL;
+
+    if ((NULL == componentName) || (NULL == objectName) || (NULL == payload) || (0 >= payloadSizeBytes))
+    {
+        OsConfigLogError(ConfigurationGetLog(), "MmiSet(%s, %s, %s, %d) called with invalid arguments", componentName, objectName, payload, payloadSizeBytes);
+        status = EINVAL;
+        return status;
+    }
+
+    if (!IsValidSession(clientSession))
+    {
+        OsConfigLogError(ConfigurationGetLog(), "MmiSet(%s, %s) called outside of a valid session", componentName, objectName);
+        status = EINVAL;
+    }
+
+    if ((MMI_OK == status) && (strcmp(componentName, g_configurationComponentName)))
+    {
+        OsConfigLogError(ConfigurationGetLog(), "MmiSet called for an unsupported component name (%s)", componentName);
+        status = EINVAL;
+    }
+
+    if ((MMI_OK == status) && (strcmp(objectName, g_desiredConfigurationObject)))
+    {
+        OsConfigLogError(ConfigurationGetLog(), "MmiSet called for an unsupported object name (%s)", objectName);
+        status = EINVAL;
+    }
+
+    if (MMI_OK == status)
+    {
+        if (NULL != (payloadString = malloc(payloadSizeBytes + 1)))

```

Comment:
+1 what? Shouldn't we not return null terminators?

Reply:
Allocates 1 more for the terminator but returns without that last byte.